frontend/test/unit/services/JsonRpcService.test.ts

```diff
@@ -0,0 +1,124 @@
+import { JsonRpcService } from '../../../src/services/JsonRpcService'
+import { IJsonRpcService } from '../../../src/services/IJsonRpcService'
+import { rpcClient } from '../../../src/utils/rpc'
+import { CallContractFunctionResult, JsonRpcResult, GetContractStateResult } from '@/types'
+import { describe, expect, it, vi, afterEach, beforeEach } from 'vitest'
+
+describe('JsonRprService', () => {
+  let jsonRpcService: IJsonRpcService
+
+  beforeEach(() => {
+    jsonRpcService = new JsonRpcService(rpcClient)
+  })
+
+  afterEach(() => {
+    vi.restoreAllMocks()
+  })
+
+  describe('getContractState', () => {
+    const mockResult: JsonRpcResult<GetContractStateResult> = {
+      data: {
+        get_have_coin: 'True',
+        id: '0x58FaA28cbAA1b52F8Ec8D3c6FFCE6f1AaF8bEEB1'
+      },
+      message: '',
+      status: 'success'
+    }
+    it('should call rpc client', async () => {
+      vi.spyOn(rpcClient, 'call').mockImplementationOnce(() => Promise.resolve(mockResult))
+      const input = {
+        contractAddress: '0x58FaA28cbAA1b52F8Ec8D3c6FFCE6f1AaF8bEEB1',
+        method: 'get_have_coin',
+        methodArguments: []
+      }
+      await jsonRpcService.getContractState(input)
+      expect(rpcClient.call).toHaveBeenCalledTimes(1)
+      expect(rpcClient.call).toHaveBeenCalledWith({
+        method: 'get_contract_state',
+        params: [input.contractAddress, input.method, []]
+      })
+    })
+
+    it('should return contract state', async () => {
+      vi.spyOn(jsonRpcService, 'getContractState').mockImplementationOnce(() =>
```

Comment:
@bradleySuira what's the value on mocking the same function that we are calling? We should mock the RPC client, call the service, and test that the client is being called with the right parameters.

Reply:
fixed